src/pages/RouteFeed.tsx

```diff
@@ -29,50 +29,49 @@ const RouteFeed = () => {
     // runs on component mount
     useEffect(() => {
         const uid = params.has('uid') ? params.get('uid') : 'none';
-        if (uid !== 'none')
-        {
+        if (uid !== 'none') {
             setRoute(getRouteById(uid!));
         }
     }, []);
 
-    useEffect( () => {
+    useEffect(() => {
         route?.getData();
-        route?.getName().then( (name) => {
+        route?.getName().then((name) => {
             setRouteName(name);
         });
-        route?.getSetter().then( (user) => {
-            user?.getUsername().then( (name) => setSetter(name));
+        route?.getSetter().then((user) => {
+            user?.getUsername().then((name) => setSetter(name));
         });
-        route?.getForum().then( (forum) => {
+        route?.getForum().then((forum) => {
             setRouteForum(forum);
         });
     }, [route]);
```

Comment:
This whole `useEffect` is a bit shady, and is probably the reason that web is throwing so many reads. `route.getData()` needs to be awaited, or else its just thrown on the call-stack with all the other ones.

```ts
useEffect(() => {
    const getData = async () => {
        await route?.getData();
        route?.getName().then(setName);
        ....   
    };

    getData();
}, [route]);
```

Reply:
Yeah this is my fault not Daniel's. This is my obtuse getData implementation from before I had a firm understanding of how the LazyObject worked, and I haven't gotten around to bringing this up to date with the new implementation flow we've been using. Now is a good time to fix this though.